src/python/pants/option/global_options.py

```diff
@@ -498,49 +507,46 @@ def register_options(cls, register):
     # global-scope options, for convenience.
     cls.register_bootstrap_options(register)
 
-    register('-x', '--time', type=bool,
-             help='Output a timing report at the end of the run.')
-    register('-e', '--explain', type=bool,
-             help='Explain the execution of goals.')
+    if register.bootstrap.v1:
+      register('-x', '--time', type=bool,
+               help='Output a timing report at the end of the run.')
+      register('-e', '--explain', type=bool,
+               help='Explain the execution of goals.')
```

Comment:
Nit: I recommend moving this into the V1 block below. Generally, it'd help readers of this file to organize this function as follows:

- options that apply to both V1 and V2
- options that apply to only V1
- options that apply to only V2

Right now they're a bit scattered throughout.